### PR TITLE
Use num::BigInt instead of i128 in constants, as constants can be of arbitrary size.

### LINKED ISSUE #140: RMC can't handle u128 constants
I tried this code:

```rust
let x : u128 = u128::MAX;
```

using the following command line invocation:

```
rmc main.rs
```

with RMC version:

I expected to see this happen: Nothing interresting.

Instead, this happened: RMC crashed, unable to handle the large constant.
```
thread 'rustc' panicked at 'called `Result::unwrap()` on an `Err` value: TryFromIntError(())', compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/expr.rs:462:30
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.54.0-dev running on x86_64-apple-darwin

note: compiler flags: -Z codegen-backend=gotoc

query stack during panic:
end of query stack
```

### PR DESCRIPTION
**Pull request by danielsn**
### Description of changes: 

Describe RMC's current behavior and how your code changes that behavior. If there are no issues this PR is resolving, explain why this change is necessary.

### Resolved issues:

Resolves #140

### Testing:

* How is this change tested? Added new regression test

* Is this a refactor change? No

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Review by markrtuttle — APPROVED** [review]

File: compiler/rustc_codegen_llvm/src/gotoc/cbmc/irep/irep.rs [hunk 1]
```diff
@@ -5,8 +5,8 @@
 use super::super::goto_program::{Location, Type};
 use super::super::MachineModel;
 use super::{IrepId, ToIrep};
+use num::BigInt;
 use std::collections::BTreeMap;
-use std::convert::TryInto;
 use std::fmt::Debug;
 
 /// The CBMC serilization format for goto-programs.
```
Comment (nchong-at-aws) [inline compiler/rustc_codegen_llvm/src/gotoc/cbmc/irep/irep.rs:12]: serialization

File: rust-tests/cbmc-reg/CodegenConstValue/u128.rs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// Avoid RUST constant propegating this away
```
Comment (nchong-at-aws) [inline rust-tests/cbmc-reg/CodegenConstValue/u128.rs:4]: propagating

File: Cargo.lock [hunk 3]
```diff
@@ -2313,21 +2313,78 @@ version = "1.0.4"
...
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "26873667bbbb7c5182d4a37c1add32cdf09f841af72da53318fdb81543c15085"
+dependencies = [
+ "num-traits",
+]
+
 [[package]]
 name = "num-integer"
-version = "0.1.43"
+version = "0.1.44"
```
Comment (nchong-at-aws) [inline Cargo.lock:2352]: Is this bump necessary for `num-bitint` 0.4.0? Will this make future upstreams trickier?

Comment (danielsn) [reply Cargo.lock:2352]: Good question. Realistically, I think we can accept any version. We can always change this if there is a problem when doing upstreaming.

**Pull request merged**